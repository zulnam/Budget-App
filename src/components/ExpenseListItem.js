import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem =  ({id, description, amount, createdAt}) => (
    <div>
        <p>{description}</p>
        <p>{amount}</p>
        <p>{createdAt}</p>
        <Link to={`/edit/${id}`} >Edit</Link>
    </div>
);

export default ExpenseListItem;