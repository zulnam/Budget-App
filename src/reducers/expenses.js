//Expenses Reducer
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        default:
            return state;
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
        case 'REMOVE_EXPENSE':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if(expense.id === action.id) {
                    // using object spread operator
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense;
                }
            });
    }
};

