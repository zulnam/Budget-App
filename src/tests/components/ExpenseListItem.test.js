import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from '../fixture/expenses';


test ('should render expense list item', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});