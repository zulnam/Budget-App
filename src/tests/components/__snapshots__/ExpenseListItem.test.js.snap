// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item 1`] = `
<div>
  <p>
    Gum
  </p>
  <p>
    195
  </p>
  <p>
    0
  </p>
  <Link
    replace={false}
    to="/edit/1"
  >
    Edit
  </Link>
</div>
`;
